vector/src/main/java/fr/gouv/tchap/features/home/room/list/TchapRoomSummaryItemFactory.kt

```diff
@@ -0,0 +1,150 @@
+                .avatarRenderer(avatarRenderer)
+                .topic(spaceChildInfo.topic)
+                .buttonLabel(stringProvider.getString(R.string.join))
+                .loading(suggestedRoomJoiningStates[spaceChildInfo.childRoomId] is Loading)
+                .memberCount(spaceChildInfo.activeMemberCount ?: 0)
+                .buttonClickListener(onJoinClick)
+    }
+
+    private fun createInvitationItem(roomSummary: RoomSummary,
+                                     changeMembershipState: ChangeMembershipState,
+                                     listener: RoomListListener?): VectorEpoxyModel<*> {
+        val secondLine = if (roomSummary.isDirect) {
+            roomSummary.inviterId
+        } else {
+            roomSummary.inviterId?.let {
+                stringProvider.getString(R.string.invited_by, it)
+            }
+        }
+
+        return RoomInvitationItem_()
+                .id(roomSummary.roomId)
+                .avatarRenderer(avatarRenderer)
+                .matrixItem(roomSummary.toMatrixItem())
+                .secondLine(secondLine)
+                .changeMembershipState(changeMembershipState)
+                .acceptListener { listener?.onAcceptRoomInvitation(roomSummary) }
+                .rejectListener { listener?.onRejectRoomInvitation(roomSummary) }
+                .listener { listener?.onRoomClicked(roomSummary) }
+    }
+
+    fun createRoomItem(
+            roomSummary: RoomSummary,
+            selectedRoomIds: Set<String>,
+            onClick: ((RoomSummary) -> Unit)?,
+            onLongClick: ((RoomSummary) -> Boolean)?
+    ): VectorEpoxyModel<*> {
+        val unreadCount = roomSummary.notificationCount
+        val showHighlighted = roomSummary.highlightCount > 0
+        val showSelected = selectedRoomIds.contains(roomSummary.roomId)
+        var latestFormattedEvent: CharSequence = ""
+        var latestEventTime: CharSequence = ""
+        val latestEvent = roomSummary.latestPreviewableEvent
+        if (latestEvent != null) {
+            latestFormattedEvent = displayableEventFormatter.format(latestEvent, roomSummary.isDirect.not())
+            latestEventTime = dateFormatter.format(latestEvent.root.originServerTs, DateFormatKind.ROOM_LIST)
+        }
+        val typingMessage = typingHelper.getTypingMessage(roomSummary.typingUsers)
+        return TchapRoomSummaryItem_()
+                .id(roomSummary.roomId)
+                .avatarRenderer(avatarRenderer)
+                .colorProvider(colorProvider)
+                // We do not display shield in the room list anymore
+                // .encryptionTrustLevel(roomSummary.roomEncryptionTrustLevel)
+                .matrixItem(roomSummary.toMatrixItem())
+                .isDirect(roomSummary.isDirect)
+                .isEncrypted(roomSummary.isEncrypted)
+                // FIXME: Update this with the logic of RoomAccessRules
+                .roomType(RoomTchapType.PRIVATE)
```

Comment:
please use your `getRoomType`, just check with @yostyle to move it to the right place (if it is not already the case)